### PR TITLE
Clang format final

### PR DESCRIPTION
**Pull request by rdaly525**
This PR updates the entire repo to consistently use a single clang-format style

Also adds a script apply_clang_format.sh to apply the clang format

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  IndentBraces:    false
+  SplitEmptyFunction: true
+  SplitEmptyRecord: true
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: BeforeColon
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: AfterColon
```
Comment (rsetaluri) [inline .clang-format:47]: Actually feel pretty strongly about this...

```
class Foo {
  Foo(int x) : 
    mem(x),
    other(false) {
...
}
```
looks really weird to me

```
class Foo {
  Foo(int x)
    : mem(x),
      other(false) {
...
}
```
 seems much better.

**Comment by rsetaluri** [comment]
also seems to be failing...

File: .clang-format (hunk 1 above)
Comment (rdaly525) [reply .clang-format:47]: Eh, that looks as weird to me. But I dont really care as long as things always line up on the next indentation level.

Comment (rsetaluri) [reply .clang-format:47]: ok, think `BeforeColon` here should do that.

Comment (rdaly525) [reply .clang-format]: Fixed.

**Comment by rdaly525** [comment]
@rsetaluri, @leonardt I think this should pass CI. Please re-review if desired.

**Pull request merged**